### DISCUSSION TITLE
N64 BlueRetro Capability Questions

### DISCUSSION THREAD

**Discussion by SwimmingKittens** [darthcloud/BlueRetro#1111] [General]
Apologies if this was posted as a question already, I did my best to search through all the discussion posts. I have a couple questions regarding the capabilities of the BlueRetro for the N64:

1. Does the detect game feature work? I couldn't get it to, I do have the UltraHDMI and 64Drive by Marshall. I was trying to set it to work with THPS and it kept saying no game detected.

2. Is there any controller detection capabilities like seeing a Switch Pro Controller or dual joycons and auto button mapping depending on which connects?

3. I think this one's a no... with 4 of these, is there a way to prioritize player one, then player 2, etc.? Right now I just have them set to manual sync except player 1.

**Comment by asweintraub** [comment]
Can you post a link to the exact hardware you're using? Some of the answers to these questions depend on it.

1. This should work if you're using the N64 firmware. If you're using the universal firmware, I think the per-game settings become per-console. I'm also not sure if game ID works on ports other than port 1.
2. Yes, see [this list](https://docs.google.com/spreadsheets/d/e/2PACX-1vT9rPK2__komCjELFpf0UYz0cMWwvhAXgAU7C9nnwtgEaivjsh0q0xeCEiZAMA-paMrneePV7IqdX48/pubhtml) for default mappings for each controller.
3. Assuming you have 4 separate dongles, then no, they can't communicate with each other. There are also power concerns with having so many dongles getting power from the controller ports at once. Have you considered using a single adapter for all controllers, such as getting a core + 4 controller cable [here](https://www.aliexpress.us/item/3256804475562445.html?pdp_npi=4%40dis%21USD%21US%20%2415.00%21US%20%247.50%21%21%2115.00%217.50%21%4021015b2417313561877646061efcec%2112000030020660749%21sh%21US%213851995000%21X&spm=a2g0o.store_pc_allItems_or_groupList.new_all_items_2007603045590.1005004661877197&gatewayAdapt=glo2usa)?

**Comment by SwimmingKittens** [comment]
NUS-001(USA) funtastic watermelon N64, UltraHDMI HW2 I believe, 64drive HW2, and a reN64 PSU w/12v USB C adapter. 

I'm not a huge fan of that aesthetic as far as that adapter, I had thought of it, but I didn't realize the complications until I tried using what I got... I was looking at this though: https://github.com/TharathielCB/BR4N64/
But it doesn't seem to be quite ready, and I don't know any modders who would be interested in an install on my hardware.

Also, I see you wrote the default mappings, but I was wondering more for custom mapping. I did update to the most recent N64 firmware and was testing on port one. I might be able to test on another cartridge and see if it's an issue with the communication between the BlueRetro and the 64drive, I just need to dig it up.

**Comment by darthcloud** [comment]
Does 64drive support GameID? I though it was abandon before that even existed. It does work with Everdrive.

For 2. Well their is a default mapping for supported controller, but you can't have a a custome mapping applied automatically base on controller specific.

**Reply by SwimmingKittens** [reply]
I appreciate the feedback. That explains a few things. Too bad about the 64drive if true. I suppose the best work around would be a decent wireless N64 controller. Any plans to support the 8bitdo n64 controller when it comes out?
